### PR TITLE
Add content/pt/docs/concepts/containers/container-lifecycle-hooks.md  

### PR DESCRIPTION
**Pull request by edsoncelio**
Changes:
* Create `content/pt/docs/concepts/containers/container-lifecycle-hooks.md `

Related to #13939

### CODE REVIEW

File: content/pt/docs/concepts/containers/container-lifecycle-hooks.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+`PostStart`
+
+Este _hook_ é executado imediatamente após um contêiner ser criado.
+Entretanto, não há garantia que o _hook_ será executado antes do ENTRYPOINT do contêiner.
+Nenhum parâmetro é passado para o manipulador.
+
+`PreStop`
+
+Esse _hook_ é chamado imediatamente antes de um contêiner ser encerrado devido a uma solicitação de API ou um gerenciamento de evento como liveness/startup probe failure, preemption, resource contention e outros.
+Uma chamada ao _hook_ `PreStop` falha se o contêiner já está em um estado finalizado ou concluído e o _hook_ deve ser concluído antes que o sinal TERM seja enviado para parar o contêiner. A contagem regressiva do período de tolerância de término do Pod começa antes que o _hook_ `PreStop` seja executado, portanto, independentemente do resultado do manipulador, o contêiner será encerrado dentro do período de tolerância de encerramento do Pod. Nenhum parâmetro é passado para o manispulador.
```
Comment (rikatz) [inline content/pt/docs/concepts/containers/container-lifecycle-hooks.md:34]: typo here -> manispulador / manipulador

File: content/pt/docs/concepts/containers/container-lifecycle-hooks.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+Uma descrição mais detalhada do comportamento de término pode ser encontrada em [Término de Pods](/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination).
+
+### Implementações de manipulador de hook
+
+Os contêineres podem acessar um _hook_ implementando e registrando um manipulador para esse _hook_.
+Existem dois tipos de manipuladores de _hooks_ que podem ser implementados para contêineres:
+
+* Exec - Executa um comando específico, como `pre-stop.sh`, dentro do cgroups e Namespaces do contêiner.
```
Comment (rikatz) [inline content/pt/docs/concepts/containers/container-lifecycle-hooks.md:43]: dentro do cgroup e namespace do container (tudo no plural ou tudo no singular)

File: content/pt/docs/concepts/containers/container-lifecycle-hooks.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+* HTTP - Executa uma requisição HTTP em um endpoint específico do contêiner.
+
+### Execução do manipulador de hook
+
+
+Quando um _hook_ de gerenciamento de ciclo de vida do contêiner é chamado, o sistema de gerenciamento do Kubernetes executa o manipulador de acordo com a ação do _hook_, `httpGet` e `tcpSocker` são executados pelo processo kubelet e `exec` é executado pelo contêiner.
```
Comment (rikatz) [inline content/pt/docs/concepts/containers/container-lifecycle-hooks.md:49]: s/tcpSocker/tcpSocket

File: content/pt/docs/concepts/containers/container-lifecycle-hooks.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+As chamadas do manipulador do _hook_ são síncronas no contexto do Pod que contém o contêiner.
+Isso significa que para um _hook_ `PostStart`, o ENTRYPOINT do contêiner e o _hook_ disparam de forma assíncrona.
+No entanto, se o _hook_ demorar muito para ser executado ou travar, o contêiner não consegue atingir o estado `running`.
+
+
+Os _hooks_ `PreStop` não são executados de forma assíncrona a partir do sinal para parar o conêiner, o _hook_ precisa finalizar a sua execução antes que o sinal TERM possa ser enviado.
```
Comment (rikatz) [inline content/pt/docs/concepts/containers/container-lifecycle-hooks.md:56]: s/conêiner/contêiner

File: content/pt/docs/concepts/containers/container-lifecycle-hooks.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+Geralmente, apenas entregas únicas são feitas.
+Se, por exemplo, um receptor de _hook_ HTTP estiver inativo e não puder receber tráfego,
+não há tentativa de reenviar.
+Em alguns casos raros, no entanto, pode ocorrer uma entrega dupla.
+Por exemplo, se um kubelet reiniciar no meio do envio de um _hook_, o _hook_ pode ser 
+reenviado depois que o kubelet voltar a funcionar.
+
+### Depurando manipuladores de _hooks_
+
+Os logs para um manipulador de _hook_ não expostos em eventos de Pod.
```
Comment (rikatz) [inline content/pt/docs/concepts/containers/container-lifecycle-hooks.md:84]: I guess the correct here would be 'são expostos em eventos de Pod"

File: content/pt/docs/concepts/containers/container-lifecycle-hooks.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+Se um manipulador falhar por algum motivo, ele transmitirá um evento.
+Para `PostStart` é o evento `FailedPostStartHook` e para `PreStop` é o evento
```
Comment (rikatz) [inline content/pt/docs/concepts/containers/container-lifecycle-hooks.md:86]: FailedPreStopHook (should start in upper case here)

**Review by rikatz — CHANGES_REQUESTED** [review]
Hi, thanks for the PR.

Left some reviews here!

File: content/pt/docs/concepts/containers/container-lifecycle-hooks.md (hunk 5 above)
Comment (edsoncelio) [reply content/pt/docs/concepts/containers/container-lifecycle-hooks.md:84]: The original is "The logs for a Hook handler are not exposed in Pod events.".  I added the missing word, thanks!

**Comment by rikatz** [comment]
/lgtm
/approve

Thank you!

**Pull request merged**